src/sonic-bgpcfgd/app/allow_list.py

```diff
@@ -0,0 +1,535 @@
+        match_community = 'match community '
+        conf = self.cfg_mgr.get_text()
+        for line in conf + [""]:
+            if inside_route_map:
+                if line.strip().startswith(match_pl_allow_list):
+                    pl_allow_list_name = line.strip()[len(match_pl_allow_list):]
+                    continue
+                elif line.strip().startswith(match_community):
+                    community_name = line.strip()[len(match_community):]
+                    continue
+                else:
+                    if pl_allow_list_name is not None:
+                        entries[route_map_seq_number] = {
+                            'pl_allow_list': pl_allow_list_name,
+                            'community': community_name,
+                        }
+                    else:
+                        if route_map_seq_number != 65535:
+                            log_warn("BGPAllowListMgr::Found incomplete route-map '%s' entry. seq_no=%d" % (route_map_name, route_map_seq_number))
+                    inside_route_map = False
+                    pl_allow_list_name = None
+                    community_name = self.EMPTY_COMMUNITY
+                    route_map_seq_number = None
+            if line.startswith(match_string):
+                found = line[len(match_string):]
+                assert found.isdigit()
+                route_map_seq_number = int(found)
+                inside_route_map = True
+        return entries
+
+    @staticmethod
+    def __find_next_seq_number(seq_numbers, has_community, route_map_name):
+        """
+        Find a next available "Allow list" route-map entry number
+        :param seq_numbers: a list of already used sequence numbers
+        :param has_community: True, if the route-map entry has community
+        :return: next available route-map sequence number
+        """
+        used_sequence_numbers = set(seq_numbers)
+        sequence_number = None
+        if has_community:  # put entries without communities after 32768
+            start_seq = 10
+            end_seq = 29990
+        else:
+            start_seq = 30000
+            end_seq = 65530
+        for i in range(start_seq, end_seq, 10):
+            if i not in used_sequence_numbers:
+                sequence_number = i
+                break
+        if sequence_number is None:
+            raise RuntimeError("No free sequence numbers for '%s'" % route_map_name)
+        info = sequence_number, "yes" if has_community else "no"
+        log_debug("BGPAllowListMgr::__find_next_seq_number '%d' has_community='%s'" % info)
+        return sequence_number
+
+    @staticmethod
+    def __restart_peers(deployment_id):
+        """
+        Restart peers with deployment_id
+        :param deployment_id: deployment_id number
+        """
+        log_info("BGPAllowListMgr::Restart peers with deployment_id=%d" % deployment_id)
+        no_error, _, _ = run_command(["vtysh", "-c", "clear bgp * soft in"])
```

Comment:
are we clearing all peers? including the upstream peers?

Reply:
it does not see you only clear the neighbor with deployment_id. impl and description does not match.

Reply:
Yes. Currently I clear all peers. But in this case I use `soft in` which will not reset any established bgp session, it will just update it.
It is not easy to find a neighbor with the deployment_id. For that I need to make a map between peer-group and neighbors and between peer-group and deployment id.

I'll look at it today.

Reply:
I am not sure I understand the impact here. does this soft in clear withdrawn and then readvertise all the routes?

Reply:
I have added implementation of reboot only required peer-groups.